structuralcodes/sections/section_integrators/_fiber_integrator.py

```diff
@@ -202,35 +241,89 @@ def integrate(
             Tuple(float, float, float): The stress resultants N, Mx and My.
         """
         # Integration over all fibers
-        x, y, F = prepared_input[0]
+        y, z, F = prepared_input[0]
 
         N = np.sum(F)
-        Mx = np.sum(F * y)
-        My = np.sum(-F * x)
+        My = np.sum(F * z)
+        Mz = np.sum(-F * y)
+
+        return N, My, Mz
+
+    def integrate_modulus(
+        self,
+        prepared_input: t.List[
+            t.Tuple[int, np.ndarray, np.ndarray, np.ndarray]
+        ],
+    ) -> NDArray:
+        """Integrate material modulus over the geometry to obtain section
+        stiffness.
+
+        Arguments:
+            prepared_input (List): The prepared input from .prepare_input().
+
+        Returns:
+            NDArray[float]: The stiffness matrix with shape (3,3).
+        """
+        # Integration over all fibers
+        y, z, MA = prepared_input[0]
 
-        return N, Mx, My
+        stiffness = np.zeros((3, 3))
+        stiffness[0, 0] = np.sum(MA)
+        stiffness[0, 1] = stiffness[1, 0] = np.sum(MA * z)
+        stiffness[0, 2] = stiffness[2, 0] = np.sum(-MA * y)
+        stiffness[1, 1] = np.sum(z * z * MA)
+        stiffness[1, 2] = stiffness[2, 1] = np.sum(-y * z * MA)
+        stiffness[2, 2] = np.sum(y * y * MA)
+
+        return stiffness
 
     def integrate_strain_response_on_geometry(
-        self, geo: CompoundGeometry, strain: ArrayLike, **kwargs
-    ):
-        """Integrate the strain response with the fiber algorithm.
+        self,
+        geo: CompoundGeometry,
+        strain: ArrayLike,
+        integrate: t.Literal['stress', 'modulus'] = 'stress',
+        **kwargs,
+    ) -> t.Tuple[t.Union[t.Tuple[float, float, float], np.ndarray], t.List]:
+        """Integrate stresses the strain response with the fiber algorithm.
```

Comment:
Suggest to change the first line in the docstring in a similar manner as above to also describe integrating modulus.

Reply:
done